web/models.py

```
@@ -2787,3 +2787,105 @@ class ScheduledPost(models.Model):
+            slug = base_slug
+            counter = 1
+            # Ensure slug is unique
+            while LearningMap.objects.filter(slug=slug).exists():
+                slug = f"{base_slug}-{counter}"
+                counter += 1
+            self.slug = slug
+
+        super().save(*args, **kwargs)
+
+    def __str__(self):
+        return f"{self.user.username}'s {self.title}"
```

Ruff (0.8.2)

Order of model's inner classes, methods, and fields does not follow the Django Style Guide: `__str__` method should come before `save` method

(DJ012)